src/main/java/microsoft/exchange/webservices/data/IAsyncResult.java

```diff
@@ -16,7 +16,7 @@
  * Represents the stauts of Asynchronous operation.
  */
 
-public interface IAsyncResult extends Future {
+public interface IAsyncResult extends Future<Object> {
```

Comment:
Thanks @serious6.  That is pretty much what I expected, Object matches everything except simple types.